tsdb/head_wal.go

```diff
@@ -620,7 +620,7 @@ func (wp *walSubsetProcessor) processWALSamples(h *Head, mmappedChunks, oooMmapp
 			if s.T <= ms.mmMaxTime {
 				continue
 			}
-			if _, chunkCreated := ms.append(s.T, s.V, 0, h.chunkDiskMapper, chunkRange); chunkCreated {
+			if _, chunkCreated := ms.append(s.T, s.V, 0, chunkRange); chunkCreated {
```

Comment:
We should also mmap the chunks _during_ a WAL replay. If the mmap chunks were removed from the disk for any reason between restart, now with this change, replay of WAL will end up holding all chunks in the memory until the replay ends. Since WAL replay already spikes the memory a bit, instead of batch mmapping later, I suggest we mmap the chunk as soon as a new one is created. Which means calling `ms.mmapHeadChunks` inside this `if` condition. (same for `appendHistogram` and `appendFloatHistogram` below)

Reply:
Missed this one, added m-map calls